packages/web-components/src/components/leadspace/leadspace.ts

```diff
@@ -0,0 +1,223 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2020
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { customElement, html, internalProperty, property, LitElement } from 'lit-element';
+import settings from 'carbon-components/es/globals/js/settings';
+import ddsSettings from '@carbon/ibmdotcom-utilities/es/utilities/settings/settings.js';
+import styles from './leadspace.scss';
+import StableSelectorMixin from '../../globals/mixins/stable-selector';
+
+const { prefix } = settings;
+const { stablePrefix: ddsPrefix } = ddsSettings;
+
+/**
+ * Leadspace type
+ */
+export enum LEADSPACE_TYPE {
+  /**
+   * Left-aligned - small style of the leadspace title (default)
+   */
+  SMALL = 'small',
+
+  /**
+   * Left-aligned - large style of the leadspace title
+   */
+  LEFT = 'left',
+
+  /**
+   * Centered type of the LeadSpace
+   */
+  CENTERED = 'centered',
+}
+
+/**
+ * Leadspace theme
+ */
+export enum LEADSPACE_THEME {
+  /**
+   * Carbon White theme (default)
+   */
+  WHITE = 'white',
+
+  /**
+   *  Carbon Gray 100 theme
+   */
+  G100 = 'g100',
+}
+
+/**
+ * The LeadSpace component.
+ *
+ * @element dds-leadspace
+ */
+@customElement(`${ddsPrefix}-leadspace`)
+class DDSLeadSpace extends StableSelectorMixin(LitElement) {
+  /**
+   * Sets background image for leadspace centered
+   */
+  protected _getBackgroundImage() {
+    return this.defaultSrc && this.type === LEADSPACE_TYPE.CENTERED ? `background-image: url(${this.defaultSrc})` : '';
+  }
+
+  /**
+   * Returns a class-name based on the gradient parameter type
+   */
+  protected _getGradientClass() {
+    return this.gradient ? `${prefix}--leadspace--gradient` : '';
+  }
+
+  /**
+   * Returns a class-name based on the type parameter type
+   */
+  protected _getTypeClass() {
+    let classname = '';
+
+    if (this.type === 'centered') {
+      classname += `${prefix}--leadspace--centered `;
+
+      if (this._hasImage) {
+        classname += `${prefix}--leadspace--centered__image `;
+      }
+    }
+
+    if (this.theme) {
+      classname += `${prefix}--leadspace--${this.theme} `;
+    }
+
+    if (this.type === 'small') {
+      classname += `${prefix}--leadspace--productive `;
+    }
+    return classname;
+  }
```

Comment:
Is it possible to make this function use `lit-element`'s `classMap`? An example can be found at: https://github.com/carbon-design-system/carbon-web-components/blob/v1.7.0/src/components/button/button.ts#L207-L214

(Same for `_getGradientClass()`)

Reply:
ah yes! this was something I was looking for! Thanks!